gobblin-utility/src/main/java/gobblin/util/filters/HiddenFilter.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.util.filters;
+
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+
+
+/**
+ * A {@link PathFilter} that filters out hidden files (those starting with '_' or '.').
+ */
+public class HiddenFilter implements PathFilter {
+
+  @Override
+  public boolean accept(Path path) {
+    return !(path.getName().startsWith("_") || path.getName().startsWith("."));
```

Comment:
I would suggest putting the offending characters into an `ImmutableSet`, like https://github.com/dnuttle/hadoop-lab/blob/master/src/main/java/net/nuttle/hadoop/HiddenPathFilter.java.